scout/adapter/mongo/hpo.py

```diff
@@ -72,22 +73,34 @@ def hpo_terms(self, query=None, hpo_term=None, limit=None):
             result(pymongo.Cursor): A cursor with hpo terms
         """
         query_dict = {}
+        search_term = None
         if query:
             query_dict = {'$or':
                 [
                     {'hpo_id': {'$regex': query, '$options':'i'}},
                     {'description': {'$regex': query, '$options':'i'}},
-                ]
+                ]   
             }
+            search_term = query
+        elif text:
+            new_string = ''
+            for i,word in enumerate(text.split(' ')):
+                if i == 0:
+                    new_string += word
+                else:
+                    new_string += ' \"{0}\"'.format(word)
+            LOG.info("Search HPO terms with %s", new_string)
+            query_dict['$text'] = {'$search': new_string}
+            search_term = text
         elif hpo_term:
             query_dict['hpo_id'] = hpo_term
+            search_term = hpo_term
 
-        if limit:
-            res = self.hpo_term_collection.find(query_dict).limit(limit)
-        else:
-            res = self.hpo_term_collection.find(query_dict)
+        limit = limit or int(10e10)
+        res = self.hpo_term_collection.find(query_dict).limit(limit).sort('hpo_number',ASCENDING)
```

Comment:
Tried to print what the query_dict looks like when I search for two terms, for instance "bladder abnormality" and "bladder abn". 
In the first case the search is the following: {'$text': {'$search': 'bladder "abnormality"'}}, and gives 1085 results
in the second case it is this one: {'$text': {'$search': 'bladder "abn"'}}, and returns just 3 results.

I believe that the multiple word search is a great improvement for the tool, but wouldn't it be even more useful if it returns everything matching both expressions? I mean the second search should return at least as many results as the first one. Would using regular expressions be much slower?